src/main/java/com/dataguise/saas/service/S3LfaService.java

```diff
@@ -0,0 +1,207 @@
+package com.dataguise.saas.service;
+
+import com.dataguise.saas.dto.DgAgentClusterDTO;
+import com.dataguise.saas.exception.BadGatewayException;
+import com.dataguise.saas.exception.BadRequestException;
+import com.dg.saas.orch.client.DgSecureRestDriver;
+import com.dg.saas.orch.exception.DgDrvException;
+import com.dg.saas.orch.models.structures.ErrorConstants;
+import com.dg.saas.orch.models.structures.Modules;
+import io.swagger.client.model.DgAgentClusterInfo;
+import io.swagger.client.model.DgCloudAgentInfo;
+import io.swagger.client.model.DgClusterInfo;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class S3LfaService {
+    private static Logger logger = Logger.getLogger(S3LfaService.class);
+
+    @Autowired
+    private DgSecureRestDriver restDriver;
+
+    /*
+This method will return the list of S3 buckets in tree format
+ */
+    public String getS3BucketsTree(String sourceId, String dirPath) throws BadGatewayException {
+        String bucketsTreeInfo = null;
+        try {
+            bucketsTreeInfo = restDriver.getS3BucketsTree(sourceId, dirPath);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        return bucketsTreeInfo;
+    }
+
+    /*
+    This method will return the list of buckets with its directories, subdirectories and files
+    */
+    public String getS3BucketsDirInfo(String sourceId, String dirPath) throws BadGatewayException {
+        String bucketsDirInfo = null;
+        try {
+            bucketsDirInfo = restDriver.getS3BucketsDirInfo(sourceId, dirPath);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        return bucketsDirInfo;
+    }
+
+    /**
+     * Method used for creationg the LFA-S3 cluster
+     * @param dgAgentClusterDTO
+     * @return
+     */
+    public Integer saveCluster(DgAgentClusterDTO dgAgentClusterDTO) throws BadGatewayException, BadRequestException{
+        Integer clusterId=null;
+        try {
+            DgClusterInfo dgClusterInfo = createClusterInfo(dgAgentClusterDTO,0);
+            clusterId = restDriver.saveClusterInfo(dgClusterInfo);
+
+        } catch (DgDrvException e) {
+            logger.error("create cluster failed " + e);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/S3LfaService.java L66
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a760202414>*